kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_lagrange_multiplier.h

```diff
@@ -0,0 +1,1182 @@
+                SparseMatrixMultiplicationUtility::AssembleSparseMatrixByBlocks(rA, matrices_p_blocks, contribution_coefficients, transpose_blocks);
+            }
+
+            // Compute LM contributions
+            TSystemVectorType b_lm(total_size_of_system);
+            ComputeRHSLMContributions(b_lm, constraint_scale_factor);
+
+            // Fill auxiliar vector
+            TSparseSpace::UnaliasedAdd(b_modified, 1.0, b_lm);
+
+            // Finally reassign
+            TSparseSpace::Copy(b_modified, rb);
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief This function is intended to be called at the end of the solution step to clean up memory storage not needed
+     */
+    void Clear() override
+    {
+        BaseType::Clear();
+
+        // Clear member variables
+        mCorrespondanceDofsSlave.clear();
+        mLagrangeMultiplierVector.resize(0,false);
+
+        // Reset flag
+        BaseType::mOptions.Set(TRANSFORMATION_MATRIX_COMPUTED, false);
+    }
+
+    /**
+     * @brief This function is designed to be called once to perform all the checks needed
+     * on the input provided. Checks can be "expensive" as the function is designed
+     * to catch user's errors.
+     * @param rModelPart The model part of the problem to solve
+     * @return 0 all ok
+     */
+    int Check(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        return BaseType::Check(rModelPart);
+
+        KRATOS_CATCH("");
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ResidualBasedBlockBuilderAndSolverWithLagrangeMultiplier";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    std::unordered_map<IndexType, IndexType> mCorrespondanceDofsSlave; /// A map of the correspondance between the slave dofs
+    TSystemVectorType mLagrangeMultiplierVector;                       /// This is vector containing the Lagrange multiplier solution
+    double mConstraintFactor = 0.0;                                    /// The constraint scale factor
+    double mAuxiliarConstraintFactor = 0.0;                            /// The auxiliar constraint scale factor
+
+    CONSTRAINT_FACTOR mConstraintFactorConsidered;                  /// The value considered for the constraint factor
```

Comment:
why the type is in capitals?

Reply:
It is a enum, enums are with capitals

Reply:
Or not?

Reply:
as i understand, the variable name should be capitals (as a variable to understand), but not the type